app/javascript/react/components/Navbar/Navbar.tsx

```diff
@@ -1,217 +1,44 @@
-import React, { ReactNode, useState } from "react";
+import React, { useRef, useState } from "react";
 import { useTranslation } from "react-i18next";
 
-import logo from "../../../../assets/images/aircasting-logo-nav.svg";
-import hamburger from "../../assets/icons/hamburger.svg";
-import habitatMapLogo from "../../assets/icons/habitatMapLogo.svg";
-import iconNavClose from "../../assets/icons/iconNavClose.svg";
-import searchIcon from "../../assets/icons/searchIcon.svg";
-import airbeamIcon from "../../assets/icons/airbeamIcon.svg";
-import goBackIcon from "../../assets/icons/goBackIcon.svg";
-import { urls } from "../../const/urls";
 import * as S from "./Navbar.style";
 
-const NavItem = ({
-  href,
-  isNavTitle,
-  isUnderline,
-  isActive,
-  children,
-}: {
-  href: string;
-  isNavTitle?: boolean;
-  isUnderline?: boolean;
-  isActive?: boolean;
-  children?: ReactNode;
-}) => {
-  return (
-    <S.ListItem $isUnderline={isUnderline} $isActive={isActive}>
-      {isNavTitle ? (
-        <S.SubNavTitleLink href={href}>{children}</S.SubNavTitleLink>
-      ) : (
-        <S.NavItemLink href={href}>{children}</S.NavItemLink>
-      )}
-    </S.ListItem>
-  );
-};
+import { LatLngLiteral, Map } from "../../types/googleMaps";
+import { DEFAULT_MAP_CENTER } from "../../const/coordinates";
+
+import DesktopHeader from "./DesktopHeader";
+import { MobileCalendarHeader, MobileHeader } from "./MobileHeader";
 
 const Navbar = () => {
   const [navMenuVisible, setNavMenuVisible] = useState(false);
+  const [location, setLocation] = useState<LatLngLiteral>(DEFAULT_MAP_CENTER);
+  const mapRef = useRef<Map>();
   const { t } = useTranslation();
 
+  const isMapPage = window.location.pathname === "/";
```

Comment:
This approach carries a risk since any URL change can disrupt this functionality. It might be worth reconsidering the design of the navbar. Instead of trying to adapt a single navbar for various views (mobile, desktop, map, non-map), we could explore the possibility of having separate navbars for each page. This isn't necessarily something to address in this PR, but rather a point to consider for future improvements.